src/server/game/Autobroadcast/AutobroadcastMgr.cpp

```diff
@@ -101,42 +133,105 @@ void AutobroadcastMgr::SendAutobroadcasts()
             weight += it->second;
             if (selectedWeight < weight)
             {
-                msg = _autobroadcasts[it->first];
+                textId = it->first;
                 break;
             }
         }
     }
     else
     {
-        msg = _autobroadcasts[urand(0, _autobroadcasts.size())];
+        textId = urand(0, _autobroadcasts.size());
     }
 
     switch (_announceType)
     {
     case AnnounceType::World:
-        SendWorldAnnouncement(msg);
+        SendWorldAnnouncement(textId);
         break;
     case AnnounceType::Notification:
-        SendNotificationAnnouncement(msg);
+        SendNotificationAnnouncement(textId);
         break;
     case AnnounceType::Both:
-        SendWorldAnnouncement(msg);
-        SendNotificationAnnouncement(msg);
+        SendWorldAnnouncement(textId);
+        SendNotificationAnnouncement(textId);
     default:
         break;
     }
 
-    LOG_DEBUG("autobroadcast", "AutobroadcastMgr::SendAutobroadcasts: '{}'", msg);
+    LOG_DEBUG("autobroadcast", "AutobroadcastMgr::SendAutobroadcasts: '{}'", textId);
 }
 
-void AutobroadcastMgr::SendWorldAnnouncement(std::string msg)
+void AutobroadcastMgr::SendWorldAnnouncement(uint8 textId)
 {
-    ChatHandler(nullptr).SendWorldTextOptional(LANG_AUTO_BROADCAST, ANNOUNCER_FLAG_DISABLE_AUTOBROADCAST, msg.data());
+    // Send localized messages to all sessions
+    ChatHandler(nullptr).DoForAllValidSessions([&](Player* player)
+        {
+            if (!player || !player->GetSession())
+            return;
+
+    // Get player's locale
+    LocaleConstant locale = player->GetSession()->GetSessionDbLocaleIndex();
+
+    std::string localizedMessage;
+
+    if (_localizedAutobroadcasts.count(textId) && _localizedAutobroadcasts[textId].count(locale))
+    {
+        // Use the localized message if available
+        localizedMessage = _localizedAutobroadcasts[textId][locale];
+    }
+    else if (_autobroadcasts.count(textId))
+    {
+        // Fallback to the default message if localization is unavailable
+        localizedMessage = _autobroadcasts[textId];
+    }
+    else
+    {
+        // Fallback message if textId doesn't exist
+        localizedMessage = "Broadcast Message Not Found";
+    }
+
+    // Send the localized or fallback message
+    ChatHandler(player->GetSession()).SendWorldTextOptional(localizedMessage, ANNOUNCER_FLAG_DISABLE_AUTOBROADCAST);
+        });
 }
 
-void AutobroadcastMgr::SendNotificationAnnouncement(std::string msg)
+void AutobroadcastMgr::SendNotificationAnnouncement(uint8 textId)
 {
-    WorldPacket data(SMSG_NOTIFICATION, (msg.size() + 1));
-    data << msg.data();
-    sWorld->SendGlobalMessage(&data);
+    for (auto const& sessionPair : sWorld->GetAllSessions())
```

Comment:
why not use DoForAllValidSessions?

Reply:
still getting a hang of all the available functions. :)